main.py

```diff
@@ -0,0 +1,109 @@
+from typedb.client import *
+
+def highestVaccines():
+
+    with session.transaction(TransactionType.READ) as read_transaction:

```

Comment:
What's the rationale behind having this be a `with` statement rather than simply storing `read_transaction` as a variable: 
```py
read_transaction = session.transaction(TransactionType.READ)
```

This question also applies to the creation of the session prior to the `while` loop.

Reply:
Thanks, its really a simplified version.
I saw that "with" statement on the TypeDB website in the documentation section.
Just out of curiosity, I wanted to ask if there is any added advantage of that?

Reply:
Apologies about the removal of the `with` statement - upon further examination I have realised that the multi-step `with` statement allows the automatic closure of clients, sessions, and transactions. This is entirely on me for lacking full understanding of the TypeDB python client, I should've checked the documentation more thoroughly.

Reply:
OK
Thanks a lot for the clarification.